build/i2c/src/lib.rs

```diff
@@ -342,10 +342,13 @@ impl ConfigGenerator {
         writeln!(
             &mut s,
             r##"
-    use drv_stm32h7_i2c::I2cController;
+    #[allow(dead_code)]
+    pub const NCONTROLLERS: usize = {ncontrollers};
+
+    use drv_stm32xx_i2c::I2cController;
 
-    pub fn controllers() -> [I2cController<'static>; {}] {{"##,
-            self.controllers.len()
+    pub fn controllers() -> [I2cController<'static>; {ncontrollers}] {{"##,
```

Comment:
Nit: now that it's defined as a constant, could we do
```suggestion
    pub fn controllers() -> [I2cController<'static>; NCONTROLLERS] {{"##,
```
?

Reply:
Agreed -- fixed